### PR TITLE
docs: mark to_lisrel docstring as raw string

### LINKED ISSUE #2249: [BUG] SyntaxWarning: invalid escape sequence '\s' (LLM generated PR)
**Describe the bug**

When I run tests I see:

```
====================================================================================== warnings summary =======================================================================================
.venv/lib/python3.12/site-packages/pgmpy/factors/base.py:80
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/factors/base.py:80: SyntaxWarning: invalid escape sequence '\s'
    """

.venv/lib/python3.12/site-packages/pgmpy/extern/tabulate.py:151
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/extern/tabulate.py:151: SyntaxWarning: invalid escape sequence '\h'
    return "\\begin{tabular}{" + tabular_columns_fmt + "}\n\hline"

.venv/lib/python3.12/site-packages/pgmpy/extern/tabulate.py:256
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/extern/tabulate.py:256: SyntaxWarning: invalid escape sequence '\['
    _invisible_codes = re.compile("\x1b\[\d*m")  # ANSI color codes

.venv/lib/python3.12/site-packages/pgmpy/extern/tabulate.py:257
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/extern/tabulate.py:257: SyntaxWarning: invalid escape sequence '\['
    _invisible_codes_bytes = re.compile(b"\x1b\[\d*m")  # ANSI color codes

.venv/lib/python3.12/site-packages/pgmpy/models/MarkovNetwork.py:611
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/models/MarkovNetwork.py:611: SyntaxWarning: invalid escape sequence '\p'
    """

.venv/lib/python3.12/site-packages/pgmpy/models/MarkovNetwork.py:725
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/models/MarkovNetwork.py:725: SyntaxWarning: invalid escape sequence '\s'
    """

.venv/lib/python3.12/site-packages/pgmpy/models/FactorGraph.py:404
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/models/FactorGraph.py:404: SyntaxWarning: invalid escape sequence '\s'
    """

.venv/lib/python3.12/site-packages/pgmpy/models/SEM.py:624
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/models/SEM.py:624: SyntaxWarning: invalid escape sequence '\g'
    """

.venv/lib/python3.12/site-packages/pgmpy/inference/ExactInference.py:763
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/inference/ExactInference.py:763: SyntaxWarning: invalid escape sequence '\s'
    """

.venv/lib/python3.12/site-packages/pgmpy/inference/mplp.py:101
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pgmpy/inference/mplp.py:101: SyntaxWarning: invalid escape sequence '\{'
    """

.venv/lib/python3.12/site-packages/pygam/pygam.py:225
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pygam/pygam.py:225: SyntaxWarning: "is not" with 'str' literal. Did you mean "!="?
    if (self.terms is not 'auto') and not (isinstance(self.terms, (TermList, Term, type(None)))):

.venv/lib/python3.12/site-packages/pygam/pygam.py:276
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pygam/pygam.py:276: SyntaxWarning: "is" with 'str' literal. Did you mean "=="?
    if self.terms is 'auto':

.venv/lib/python3.12/site-packages/pygam/core.py:57
  /home/danbolser/Work/nomaly-disease-browser/.venv/lib/python3.12/site-packages/pygam/core.py:57: SyntaxWarning: "is not" with 'str' literal. Did you mean "!="?
    if flatten_attrs and k is not 'terms':
```

**To Reproduce**

The GitHub Action already shows these warnigs, e.g. here:
https://github.com/pgmpy/pgmpy/actions/runs/16107237331/job/45444750390

**Expected behavior**

No warnings I guess?

**Additional context**

I'm learning, so please feel free to slap me.

**Versions**

`uv pip freeze`

<details>
aiohappyeyeballs==2.6.1
aiohttp==3.12.13
aiosignal==1.4.0
annotated-types==0.7.0
anyio==4.9.0
asttokens==3.0.0
attrs==25.3.0
black==25.1.0
certifi==2025.6.15
cfgv==3.4.0
charset-normalizer==3.4.2
click==8.2.1
contourpy==1.3.2
coverage==7.9.2
cycler==0.12.1
daft-pgm==0.1.4
decorator==5.2.1
distlib==0.3.9
distro==1.9.0
execnet==2.1.1
executing==2.2.0
filelock==3.18.0
fonttools==4.58.5
frozenlist==1.7.0
fsspec==2025.5.1
h11==0.16.0
hf-xet==1.1.5
httpcore==1.0.9
httpx==0.28.1
huggingface-hub==0.33.2
identify==2.6.12
idna==3.10
importlib-metadata==8.7.0
iniconfig==2.1.0
ipython==9.4.0
ipython-pygments-lexers==1.1.1
jedi==0.19.2
jinja2==3.1.6
jiter==0.10.0
joblib==1.5.1
jsonschema==4.24.0
jsonschema-specifications==2025.4.1
kiwisolver==1.4.8
litellm==1.61.15
markupsafe==3.0.2
matplotlib==3.10.3
matplotlib-inline==0.1.7
mock==5.2.0
mpmath==1.3.0
multidict==6.6.3
mypy-extensions==1.1.0
networkx==3.5
nodeenv==1.9.1
numpy==2.3.1
nvidia-cublas-cu12==12.6.4.1
nvidia-cuda-cupti-cu12==12.6.80
nvidia-cuda-nvrtc-cu12==12.6.77
nvidia-cuda-runtime-cu12==12.6.77
nvidia-cudnn-cu12==9.5.1.17
nvidia-cufft-cu12==11.3.0.4
nvidia-cufile-cu12==1.11.1.6
nvidia-curand-cu12==10.3.7.77
nvidia-cusolver-cu12==11.7.1.2
nvidia-cusparse-cu12==12.5.4.2
nvidia-cusparselt-cu12==0.6.3
nvidia-nccl-cu12==2.26.2
nvidia-nvjitlink-cu12==12.6.85
nvidia-nvtx-cu12==12.6.77
openai==1.93.1
opt-einsum==3.4.0
packaging==25.0
pandas==2.3.1
parso==0.8.4
pathspec==0.12.1
patsy==1.0.1
pexpect==4.9.0
-e file:///home/dan/Build/pgmpy
pillow==11.3.0
platformdirs==4.3.8
pluggy==1.6.0
pre-commit==4.2.0
prompt-toolkit==3.0.51
propcache==0.3.2
ptyprocess==0.7.0
pure-eval==0.2.3
pydantic==2.11.7
pydantic-core==2.33.2
pygments==2.19.2
pygraphviz==1.14
pyparsing==3.2.3
pyro-api==0.1.2
pyro-ppl==1.9.1
pytest==8.4.1
pytest-cov==6.2.1
pytest-xdist==3.8.0
python-dateutil==2.9.0.post0
python-dotenv==1.1.1
pytz==2025.2
pyyaml==6.0.2
referencing==0.36.2
regex==2024.11.6
requests==2.32.4
rpds-py==0.26.0
scikit-learn==1.7.0
scipy==1.16.0
setuptools==80.9.0
six==1.17.0
sniffio==1.3.1
stack-data==0.6.3
statsmodels==0.14.5
sympy==1.14.0
threadpoolctl==3.6.0
tiktoken==0.9.0
tokenize-rt==6.2.0
tokenizers==0.21.2
torch==2.7.1
tqdm==4.67.1
traitlets==5.14.3
triton==3.3.1
typing-extensions==4.14.1
typing-inspection==0.4.1
tzdata==2025.2
urllib3==2.5.0
virtualenv==20.31.2
wcwidth==0.2.13
xdoctest==1.2.0
xgboost==3.0.2
yarl==1.20.1
zipp==3.23.0

</details>

### PR DESCRIPTION
**Pull request by dbolser**
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
- Fixes https://github.com/pgmpy/pgmpy/issues/2249

### List of changes to the codebase in this pull request

- Converted several docstrings that contained LaTeX sequences to raw strings to avoid invalid escape warnings. For example, the docstring of factor_sum_product now begins with r""" so that sequences like \sum are handled correctly
- Corrected LaTeX table generation to properly escape backslashes in _latex_line_begin_tabular
- Updated the partition function documentation in FactorGraph with a raw string to ensure the math expression renders without warnings
- Added raw string prefixes to documentation in DiscreteMarkovNetwork so expressions such as {X \perp ...} are valid
- Fixed similar issues in ExactInference, Mplp, and SEM by applying raw strings to their docstrings

### CODE REVIEW

**Comment by dbolser** [comment]
Moare context:

Here are the file‑level changes introduced by the commit 556bd74d:

- pgmpy/factors/base.py
  - The docstring for factor_sum_product now uses a raw string to avoid escape warnings
- pgmpy/inference/ExactInference.py
  - The _is_converged method’s docstring is marked as raw for correct LaTeX handling
- pgmpy/inference/mplp.py
  - The Cluster inner class docstring is switched to a raw string, and its parameter description for cluster_potential has been reindented
- pgmpy/models/DiscreteMarkovNetwork.py
  - Both get_local_independencies and get_partition_function are documented with raw strings
- pgmpy/models/FactorGraph.py
  - The docstring of get_partition_function now begins with r"""
- pgmpy/models/SEM.py
  - to_lisrel and __standard_lisrel_masks docstrings were updated to raw strings to properly display LaTeX and other escapes

These adjustments are primarily focused on preventing SyntaxWarning: invalid escape sequence warnings by converting relevant docstrings into raw strings. The code functionality remains unchanged.

**Review by ankurankan — APPROVED** [review]
Thanks for the PR @dbolser. Looks good to me.
